pkg/gds/emails/mock.go

```diff
@@ -0,0 +1,85 @@
+package emails
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/sendgrid/rest"
+	sgmail "github.com/sendgrid/sendgrid-go/helpers/mail"
+)
+
+var MockEmails [][]byte
+
+func PurgeMockEmails() {
+	MockEmails = nil
+}
+
+type mockSendGridClient struct{}
+
+func (c *mockSendGridClient) Send(msg *sgmail.SGMailV3) (rep *rest.Response, err error) {
```

Comment:
This is really truly excellent! My first assumption was that you went through the `sendgrid` code base to mimic the validation in the `Send` method of the actual client. However, when I went to that method to take a look, I realized how much more work you actually did! I found the `GetRequestBody` function call from the helpers, but my quick look did not see the chain to do all of the email address validation - really, really well done! 

And this was super important, if we didn't do any email validation then we wouldn't be able to capture possible errors that sendgrid might return to us - you've given us not only address validation but also created a mechanism with which we can validate other parts of the request if needed. Thank you!